src/game-logic/cows/cow-selection.test.js

```diff
@@ -0,0 +1,113 @@
+import { act, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import { farmhandStub } from '../../test-utils/stubs/farmhandStub'
+import { generateCow, getCowDisplayName } from '../../utils'
+import { stageFocusType } from '../../enums'
+
+describe('cow selection', () => {
+  const cowId1 = 'foo'
+  const cowId2 = 'bar'
+  const cowStub1 = generateCow({ id: cowId1 })
+  const cowStub2 = generateCow({ id: cowId2 })
+  let cow1 = null
+  let cow2 = null
+
+  const getCow1 = async () => {
+    return (
+      await screen.findByAltText(getCowDisplayName(cowStub1, cowId1, false))
+    ).closest('.cow')
+  }
+
+  const getCow2 = async () => {
+    return (
+      await screen.findByAltText(getCowDisplayName(cowStub2, cowId2, false))
+    ).closest('.cow')
+  }
+
+  beforeEach(async () => {
+    jest.useFakeTimers()
+
+    cow1 = null
+    cow2 = null
+
+    await farmhandStub({
+      cowInventory: [cowStub1, cowStub2],
+      stageFocus: stageFocusType.COW_PEN,
+      purchasedCowPen: 1,
+
+      // Usually this gets initialized during the boot process, but it needs to
+      // be stubbed to prevent a spurious crash
+      cowForSale: generateCow(),
+      hasBooted: true,
+    })
+
+    cow1 = await getCow1()
+    cow2 = await getCow2()
+  })
+
+  afterEach(() => {
+    // Allow all cow movement timers to expire
+    act(() => {
+      jest.advanceTimersByTime(2000)
+    })
+  })
+
+  test('no cows are selected by default', () => {
+    expect(cow1.classList).not.toContain('is-selected')
+    expect(cow2.classList).not.toContain('is-selected')
+  })
+
+  test('a cow can be selected', () => {
+    userEvent.click(cow1)
+
+    act(() => {
+      // Prevent async update warning from Popper.js (used by Material UI
+      // tooltips). See "Case 2" here:
+      // https://davidwcai.medium.com/react-testing-library-and-the-not-wrapped-in-act-errors-491a5629193b
+      jest.advanceTimersByTime(500)
+    })
+
+    expect(cow1.classList).toContain('is-selected')
+    expect(cow2.classList).not.toContain('is-selected')
```

Comment:
i don't feel this line needs to be here. i get what you're thinking, but it feels a little misleading since in reality CowPen will have N number of items, and just testing it checking these 2 statically isn't the same to me. i could get behind an approach that uses a loop to check that of all rendered cows no more than 1 is ever selected, but being as your next test is the deselect version, the only expectation that matters to me is what happens to the one we clicked on in this test (so this note applies to that test as well). if this didn't make sense let me know cause i feel like i used a lot of words, hah